### PR TITLE
Prepare tutorial

### PR DESCRIPTION
**Pull request by junha1**

### CODE REVIEW

**Review by majecty — CHANGES_REQUESTED** [review]

File: coordinator/src/module.rs [hunk 1]
```diff
@@ -22,57 +36,125 @@ use ctypes::{ChainParams, CompactValidatorSet};
 use remote_trait_object::{service, Service, ServiceRef};
 use serde::{Deserialize, Serialize};
 
+/// An identifier of the session.
```
[the remark this answers is not in the document]
Comment (junha1) [reply coordinator/src/module.rs:39]: What are 'other similar comments'?

File: foundry/main.rs [hunk 2]
```diff
@@ -108,4 +108,12 @@ mod timestamp_setup {
             Arc::new(start::<<SingleProcess as ExecutionScheme>::Ipc, codechain_timestamp::sorting::Module>),
         );
     }
+
+    #[distributed_slice(MODULE_INITS)]
+    fn util() {
+        add_function_pool(
+            "a060000000012345678901234567890123456789012345678901234567890123".to_owned(),
```
[the remark this answers is not in the document]
Comment (junha1) [reply foundry/main.rs:115]: I just set any value.

File: keystore/src/bin/sign.rs [hunk 3]
```diff
@@ -0,0 +1,17 @@
+use ckey::Ed25519Private as Private;
+
+fn main() -> Result<(), String> {
+    let args: Vec<String> = std::env::args().collect();
+    if args.len() != 3 {
+        return Err(format!("Wrong number of arguments; Expected 2, but given {}", args.len()));
+    }
+
+    let private = Private::from_slice(&hex::decode(&args[1])
```
[the remark this answers is not in the document]
Comment (junha1) [reply keystore/src/bin/sign.rs]: Ok, but note that we will eventually use a web service to sign txes

File: coordinator/src/module.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (majecty) [reply coordinator/src/module.rs:39]: For example, "A service to initialize the genesis state","Sets up the genesis state"

[the remark this answers is not in the document]
Comment (junha1) [reply coordinator/src/module.rs]: What is the cost of having them? I know that they are not very informative, but I think it helps anyway. It is a 'service', and it 'sets up' (not signal, query, get prepared or something).

[the remark this answers is not in the document]
Comment (junha1) [reply coordinator/src/module.rs]: And considering that this interface is for public (not only for Foundry developers), I think it's better to keep doc comments as complete as possible.

File: integration-test/src/lib.rs [hunk 4]
```diff
@@ -15,7 +15,6 @@
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
 use awc::Client;
```
[the remark this answers is not in the document]
Comment (junha1) [reply integration-test/src/lib.rs:17]: done

File: timestamp/src/util/mod.rs [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+struct GraphQlRoot {}
+
+pub struct GraphQlRequestHandler {
+    /// A runtime to process the asynchronous result of the query
+    tokio_runtime: tokio::runtime::Runtime,
+}
+
+#[async_graphql::Object]
+impl GraphQlRoot {
+    async fn sign_and_encode_tx(&self, private: String, content: String) -> async_graphql::FieldResult<String> {
```
[the remark this answers is not in the document]
Comment (junha1) [reply timestamp/src/util/mod.rs:34]: done

File: coordinator/src/module.rs [hunk 6]
```diff
@@ -14,6 +14,20 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
+//! A set of service traits expected to be implemented and exported by Foundry modules.
+//!
+//! Most of methods take a `SessionID` to specify which session to use.
+//! Session is a identifier which represents a state at certain height of block.
```
[the remark this answers is not in the document]
Comment (junha1) [reply coordinator/src/module.rs]: fixed

File: keystore/src/bin/sign.rs [hunk 7]
```diff
@@ -0,0 +1,17 @@
+use ckey::Ed25519Private as Private;
+
+fn main() -> Result<(), String> {
+    let args: Vec<String> = std::env::args().collect();
+    if args.len() != 3 {
+        return Err(format!("Wrong number of arguments; Expected 2, but given {}", args.len()));
```
[the remark this answers is not in the document]
Comment (junha1) [reply keystore/src/bin/sign.rs]: fixed

**Review by majecty — APPROVED** [review]

**Pull request merged**